lib/bots/notifybot.ts

```diff
@@ -0,0 +1,918 @@
+	public code: number;
+	/** Owner of the repo. */
+	public owner: string;
+	/** Name of the repo. */
+	public repo: string;
+	/** Version of the component being operated upon. */
+	public version: string;
+	/** Type of the error. */
+	public type = 'NotifyBotError';
+
+	/**
+	 * Constructs a new NotifyBot error.
+	 * @param code     NotifyBotError code to set.
+	 * @param message  Message to send back as reason error occurred.
+	 */
+	constructor(code: number, message: string, owner: string, repo: string, version: string) {
+		super();
+
+		// Set from parameters.
+		this.code = code;
+		this.message = message;
+		this.owner = owner;
+		this.repo = repo;
+		this.version = version;
+	}
+}
+
+/** Interface for storing the PRs for each version of a component. */
+interface VersionPRTracker {
+	[version: string]: string[];
+}
+
+/** Tracks a component from the keyframe file. */
+interface ComponentTracker {
+	/** Name of the component. */
+	name: string;
+	/** Old version (if any) of the component from previous iteration of keyframe file. */
+	oldVersion: string | null;
+	/** New version of component from current iteration of keyframe file. */
+	newVersion: string;
+	/** Full repository name of the component. */
+	repo: string;
+	/** Whether the version is incrementing or decrementing. */
+	incrementing: boolean;
+	/** All the PRs for each version of the component.  */
+	versions?: VersionPRTracker;
+}
+
+/** Configuration interface for initialising NotifyBot. */
+export interface NotifyBotConfig {
+	/** The name of the bot. */
+	botName: string;
+	/** The Github App ID to use. */
+	githubApp: string;
+	/** The PEM for communicating with Github. */
+	githubPEM: string;
+	/** The secret webhook key for validating messages from Github. */
+	githubSecret: string;
+	/** The API key for communicating with Front. */
+	frontApiKey: string;
+	/** The username to use in Front comments (this should be a unique username for NotifyBot use). */
+	frontUser: string;
+}
+
+/** Stores Front topic information relating to component issues for a PR. */
+interface TopicIssues {
+	/** A full description of each issue along with their location in the component repository. */
+	relatedIssues: string[];
+	/** The Front topic IDs that should be commented on for these issues. */
+	topics: string[];
+}
+
+/** The port that Github attempts to send events on. */
+const NotifyBotPort = 8399; // Not a listed registered port
+
+/**
+ * The RegExp to use to determine if a PR/Issue is connected to another. See the following links:
+ *   * https://help.github.com/articles/closing-issues-via-commit-messages/
+ *   * https://help.waffle.io/faq/waffle-workflow/use-waffles-connect-keyword-to-connect-prs-to-issues
+ * Additionally, this searches for the 'Connects-To:` keyword, which is specified by the Commit/PR guidelines.
+ */
+const IssueRefRE = new RegExp('(?:close[sd]?:?|fix(?:e[sd]{1})?:?|resolve[sd]?:?|' +
+	'connect(?:s|ed)?[\\s]+(?:to)?|connects-to?:?)[\\s]+(#[0-9]+)', 'i');
```

Comment:
Yes, very intentional. The NB spec. says it will only match against local issues and those from HQ with the `hq:` tag.